### PR TITLE
mvcc: add simple benchmark for scan

### PR DESCRIPTION
**Pull request by c4pt0r**
go test -check.b testMvccBenchSuite
testMvccBenchSuite.BenchmarkScan1000Keys10Versions10Bytes    100 28129059 ns/op
testMvccBenchSuite.BenchmarkScan100Keys10Versions10Bytes    500  4140501 ns/op
testMvccBenchSuite.BenchmarkScan100Keys1Versions10Bytes   1000  1058576 ns/op
testMvccBenchSuite.BenchmarkScan10Keys10Versions10Bytes   2000   613414 ns/op
testMvccBenchSuite.BenchmarkScan10Keys1Versions10Bytes  10000   119237 ns/op
testMvccBenchSuite.BenchmarkScan1Keys1Versions10Bytes  50000    23190 ns/op

### CODE REVIEW

**Comment by coocood** [comment]
Any clean up work, like delete the data has been written?

How about load data in `SetUpSuite` and delete them in `TearDownSuite`?

File: store/localstore/mvcc_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	c.ResetTimer()
+	for i := 0; i < c.N; i++ {
+		keyIdx := rand.Int31n(int32(numKeys - scanKeys))
+		key := codec.EncodeUintDesc(buf[:4], uint64(keyIdx))
+		it, _ := s.Seek(key)
+		for j := 0; j < scanKeys; j++ {
+			it.Next()
+		}
+	}
+	c.StopTimer()
```
Comment (xiang90) [inline store/localstore/mvcc_bench_test.go]: you actually do not need this.

File: store/localstore/mvcc_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	// try scan fixed keys in 100000 keys db.
+	numKeys := 100000
+	t.loadTestData(100000, numVersions, valSize)
+	// minimize the cost of constructing keys
+	buf := append(make([]byte, 0, 64), []byte("row_")...)
+	// TODO(dongxu): use random version
+	s := &dbSnapshot{t.s, kv.NewVersion(11)}
+	c.ResetTimer()
+	for i := 0; i < c.N; i++ {
+		keyIdx := rand.Int31n(int32(numKeys - scanKeys))
```
Comment (xiang90) [inline store/localstore/mvcc_bench_test.go]: line 89 and 90 seems to cost a lot. probably you also want to prepare this before start the actual benchmark loop.

File: store/localstore/mvcc_bench_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	store, err := d.Open(path)
+	if err != nil {
+		panic(err)
+	}
+	return store
+}
+
+var randLetters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+
+func randBytes(r *rand.Rand, size int) []byte {
```
Comment (xiang90) [inline store/localstore/mvcc_bench_test.go]: if we only want randBytes, https://golang.org/pkg/crypto/rand/#Read is simpler. If we do want rand letter, probably we should rename randBytes to randLetters.

File: store/localstore/mvcc_bench_test.go (hunk 2 above)
Comment (c4pt0r) [inline store/localstore/mvcc_bench_test.go]: Actually, I don't think I could prepare the keys outside benchmark loop, I just want to bench the random seek/scan performance.
Or maybe we can prepare a big random index array outside, just like this:

```
for i :=0 ; i < 100000; i++ {
    idx[i] = random()
}
for i :=0; i < c.N; i++ {
    seek(idx[i])
}
```

File: store/localstore/mvcc_bench_test.go (hunk 3 above)
Comment (c4pt0r) [inline store/localstore/mvcc_bench_test.go]: good point.

File: store/localstore/mvcc_bench_test.go (hunk 2 above)
Comment (xiang90) [inline store/localstore/mvcc_bench_test.go]: @c4pt0r I guess this commit is updated so the line changed. :) I should be more clear.

I mean
```go
keyIdx := rand.Int31n(int32(numKeys - scanKeys))
key := codec.EncodeUintDesc(buf[:4], uint64(keyIdx))
```

>  Or maybe we can prepare a big random index array outside, just like this?

So, yes. I think this makes the benchmark more accurate. Thanks!

**Comment by c4pt0r** [comment]
@shenli  PTAL

**Pull request closed without merging**